src/main/java/com/example/springresttask/controller/FileController.java

```diff
@@ -0,0 +1,43 @@
+package com.example.springresttask.controller;
+
+import com.example.springresttask.model.UploadedFile;
+import com.example.springresttask.service.FileService;
+import lombok.AccessLevel;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.FieldDefaults;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.util.List;
+
+
+@RestController
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+@RequiredArgsConstructor
+@RequestMapping("/files")
+public class FileController {
+    FileService fileService;
+
+    @GetMapping
+    public List<UploadedFile> files() {
+        return fileService.getFileList();
+    }
+
+    @PostMapping("/upload")
+    public ResponseEntity uploadFile(@RequestParam("file") MultipartFile file) {
+        return fileService.uploadFile(file);
+    }
+
+    @PostMapping("/update")
+    public ResponseEntity updateFile(@RequestParam("newFileName") String newFileName,
```

Comment:
If the request param is the same as the method's parameter, you can omit the name in `@RequestParam`